lib/push.js

```diff
@@ -5,26 +5,45 @@ const fs = require('fs')
 const { logger } = require('./utils/log-handler')
 const { checkDpIsThere, getToken } = require('./utils/common')
 const { spinner } = require('./utils/tools')
+const { addResource, writeDp } = require('./init')
 const path = require('path')
 const request = require('request-promise-native')
 const urljoin = require('url-join')
+const Datapackage = require('datapackage').Datapackage
 
 
-const push = async() => {
-  spinner.start()
+const push = async(filePath) => {
+  let dpjson
+  // get configs
   const config = creds.readConfig()
   if (!config) {
     logger('Config file not found. Setup configurations by running "data config" command.', 'abort', true, spinner)
   }
-  if (!checkDpIsThere()) {
-    logger('datapckage.json not found!', 'abort', true, spinner)
+  // get dpjson
+  if(!filePath) {
+    if (!checkDpIsThere()) {
+      logger('datapckage.json not found!', 'abort', true, spinner)
+    }
+    dpjson = JSON.parse(fs.readFileSync('datapackage.json').toString())
+  } else {
+    // prepare temp descriptor for the single file push
+    let descriptor = {
+        name: filePath.replace(/^.*[\\\/]/, ''),
+        resources: []
+    }
+    const dpObj = await new Datapackage(descriptor)
+    await addResource(filePath, dpObj)
```

Comment:
Think will be good to start spinner here as well. With message "preparing data" or just "preparing". Cause what if file is really big